contracts/Lock.sol

```diff
@@ -0,0 +1,33 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.9;
+
+// Uncomment this line to use console.log
+// import "hardhat/console.sol";
+
+contract Lock {
+    uint public unlockTime;
+    address payable public owner;
+
+    event Withdrawal(uint amount, uint when);
+
+    constructor(uint _unlockTime) payable {
+        require(block.timestamp < _unlockTime, "Unlock time should be in the future");
```

Comment:
I would like this template to use [custom errors](https://blog.soliditylang.org/2021/04/21/custom-errors/) instead of revert reason strings. Could you please refactor this to use custom errors instead?

Reply:
OK, I'll do it.